src/storage/kv.rs

```diff
@@ -97,7 +108,25 @@ impl LogKeyValueStore {
         key: &KVKey,
         transaction_id: Option<TransactionId>,
     ) -> KVResult<Option<KVValue>> {
-        match self.key_pointer_map.get(&key) {
+        let key_pointer_map = if let Some(transaction_id) = transaction_id {
+            let transaction_meta_data = self
+                .transaction_manager
+                .get_transaction_meta_data(&transaction_id)?;
+            transaction_meta_data.snapshot
+        } else {
+            self.key_pointer_map.clone()
+        };
+
+        self.read_from_snapshot(&key_pointer_map, key, transaction_id)
+    }
+
+    fn read_from_snapshot(
```

Comment:
Should this be a method on `Snapshot`? i.e. `snapshot.read(&key, &tid)`.

Reply:
Snapshot is just a alias of the hashmap, it is not a struct.

`pub type Snapshot = HashMap<KVKey, HashMap<Option<TransactionId>, LogPointer>>`

Reply:
Yeah I understand that, but it can be made a newtype, something like `struct Snapshot(HashMap<KVKey, HashMap<Option<TransactionId>, LogPointer>>)`